docs/src/main/tut/guidelines.md

```diff
@@ -69,6 +69,15 @@ The user doesn't need to specify the type `A` which is given by the parameter.
 You probably noticed that there is a `val dummy: Boolean` in the `PurePartiallyApplied` class. This is a trick we used
 to make this intermediate class a [Value Class](http://docs.scala-lang.org/overviews/core/value-classes.html) so that there is no cost of allocation, i.e. at runtime, it doesn't create an instance of `PurePartiallyApplied`. We also hide this partially applied class by making it package private and placing it inside an object.
 
+### <a id="ops-classes" href="#implicit-naming"></a> Implicit naming
 
-#### TODO: 
+In a widely-used library it's important to minimize the chance that the names of implicits will be used by others and
+therefore name our implicits according to the following rules:
+
+- Implicits should start with "cats" -- a third-party will less likely do this.
+- The type and the type class should be mentioned in the name.
+
+As an example, for `Monoid[List[A]]`, the name of the implicit should be `catsMonoidForList`.
```

Comment:
Thanks so much! We actually also include the package name in the name now. 
`catsPackageNameTypeclassNameForDataStruct`

Reply:
Thank you for letting me know. Would the following be correct ?

In a widely-used library it's important to minimize the chance that the names of implicits will be used by others and therefore name our implicits according to the following rules:

- Implicits should start with "cats" followed by the package name.
- The type and the type class should be mentioned in the name.

As an example, for `cats.kernel.Monoid[List[A]]`, the name of the implicit should be `catsKernelMonoidForList`.

Reply:
We should probably mention that in `cats.instances` we don't follow this rule and use `catsStd....`.

Reply:
The package used in the name is package where the instance is defined, your example gives the impression that it's the package where the type class is defined. 
BTW, we are not really strictly following this rule, for instance, in `cats.instances` we named an instances as 
`catsStdInstancesForOption`
https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/instances/option.scala#L8

here `Std` is to represent the package `instances` which used to be called `std`, we didn't change the name here because `catsInstancesInstancesForOption` would be weird.
the next word `Instances` in the name is used here because this instance is an instance for multiple type classes. 
So we do have some flexibility here.

Reply:
My recommendation would be just 
```
In a widely-used library it's important to minimize the chance that the names of implicits will be used by others and therefore name our implicits according to the following rules:

* Implicits should start with "cats" followed by the package (where instance is defined) name.
* The type and the type class should be mentioned in the name.
* If the instance is for multiple type classes, use `InstancesFor` instead 
* This rule is relatively flexible. Use what you see appropriate. The goal is to maintain the uniqueness and avoid conflicts. 
```

Reply:
It seems like Std is used when the instance is for a Scala standard library type.

Reply:
I've gone through all of the instances with @tbje and as far as I can tell the current revision is correct in all cases.